### PR TITLE
chore: bump angular from 12.x to 13

### LINKED ISSUE #7044: V13 update
- [x] Upgrade to angular v13
- [x] Using Ivy to build
- [x] Remove deprecated APIs and add migration schematics
- [ ] #7009 @wendellhu95 

Note:
This should be merged after styles ant-design synced #7099.

### PR DESCRIPTION
**Pull request by simplejason**
close #7044 

## TODO

- [x] Upgrade to angular v13 ( [Angular Update Guide](https://update.angular.io/?l=3&v=12.0-13.0) )
- [x] Upgrade @angular/cdk to v13
- [x] Testing Ivy to build & tests ( [CLI Issue](https://github.com/angular/angular-cli/issues/19070) | [NG3003 Error](https://angular.io/errors/NG3003) )
- [ ] ~~Update official document~~
- [x] update icons to v13 @wendellhu95 , latest @13.0.0-beta.0  

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,39 +25,39 @@
   },
   "dependencies": {
     "@angular/cdk": "^12.1.0",
-    "@ant-design/icons-angular": "^12.0.3",
+    "@ant-design/icons-angular": "^13.0.0-beta.0",
     "date-fns": "^2.16.1",
     "ngx-hover-preload": "0.0.3",
     "resize-observer-polyfill": "^1.5.1"
```
Comment (wendellhu95) [inline package.json:31]: Consider use https://github.com/juggle/resize-observer instead cuz resize-observer-polyfill seems deserted. Or drop this polyfill cuz Angular v13 dropped support for IE11.

Comment (simplejason) [reply package.json:31]: I think we can remove it because it is only used for IE, right?

**Comment by wendellhu95** [comment]
@simplejason I updated icons in commt eb352ef.

**Comment by qtopie** [comment]
Just provide some useful links, pls check [this related PR](https://github.com/angular/angular-cli/commit/21809e14cd5c666c82fdaebc9e601341dfb76d0a) and [solved issue](https://github.com/angular/angular-cli/issues/11051) of Angular CLI. Hope Angular 13 will be supported soon.

**Comment by boralbgra** [comment]
Hi! I can't install Ng Zorro in my angular project.

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: example@0.1.0
npm ERR! Found: @angular/animations@13.0.3
npm ERR! node_modules/@angular/animations
npm ERR!   @angular/animations@"~13.0.3" from the root project       
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/animations@"^12.1.0" from ng-zorro-antd@12.1.0
npm ERR! node_modules/ng-zorro-antd
npm ERR!   ng-zorro-antd@"12.1.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

**Comment by simplejason** [comment]
> Hi! I can't install Ng Zorro in my angular project.
> 
> npm ERR! code ERESOLVE npm ERR! ERESOLVE unable to resolve dependency tree npm ERR! npm ERR! While resolving: example@0.1.0 npm ERR! Found: @angular/animations@13.0.3 npm ERR! node_modules/@angular/animations npm ERR! @angular/animations@"~13.0.3" from the root project npm ERR! npm ERR! Could not resolve dependency: npm ERR! peer @angular/animations@"^12.1.0" from ng-zorro-antd@12.1.0 npm ERR! node_modules/ng-zorro-antd npm ERR! ng-zorro-antd@"12.1.0" from the root project npm ERR! npm ERR! Fix the upstream dependency conflict, or retry npm ERR! this command with --force, or --legacy-peer-deps npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

ng-zorro-antd@13.0.0 is not released now

**Comment by maratkalibek** [comment]
I'm curious about the release with Angular 13 support. Is this PR ready to be merged? @simplejason are you waiting only for approval? Or more work to be done in this PR?

**Comment by simplejason** [comment]
> I'm curious about the release with Angular 13 support. Is this PR ready to be merged? @simplejason are you waiting only for approval? Or more work to be done in this PR?

It's ready to merge now, only wait to review, I think v13 will be coming soon in this week.

**Comment by simplejason** [comment]
@hsuanxyz @yangjunhan @wendellhu95 @wenqi73 Hi all, the changes about v13 have been committed, if no request changes are suggested, we can merge this PR to master today. And then we need to prepare the upgrade documentation to ensure that users can successfully upgrade from v12 to v13.

File: components/core/overlay/nz-connected-overlay.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ export class NzConnectedOverlayDirective {
   }
 
   private updateArrowPosition(position: ConnectedOverlayPositionChange): void {
-    const originEl = this.cdkConnectedOverlay.origin.elementRef.nativeElement as HTMLElement;
+    const originEl = (this.cdkConnectedOverlay.origin as ElementRef).nativeElement as HTMLElement;
```
Comment (hsuanxyz) [inline components/core/overlay/nz-connected-overlay.ts:39]: Could this be another types?

File: components/form/form-control.component.ts [hunk 3]
```diff
@@ -134,7 +134,7 @@ export class NzFormControlComponent implements OnChanges, OnDestroy, OnInit, Aft
     this.validateChanges.unsubscribe();
     /** miss detect https://github.com/angular/angular/issues/10887 **/
     if (this.validateControl && this.validateControl.statusChanges) {
-      this.validateChanges = this.validateControl.statusChanges
+      this.validateChanges = (this.validateControl.statusChanges as Observable<string>)
```
Comment (hsuanxyz) [inline components/form/form-control.component.ts:137]:
```suggestion
      this.validateChanges = (this.validateControl.statusChanges as Observable<SafeAny>)
```

File: scripts/site/generate-site.js [hunk 4]
```diff
@@ -41,7 +41,7 @@ function generate(target) {
       }
     }
     const componentDirPath = path.join(rootPath, componentName);
-    const skips = ['style', 'core', 'locale', 'cdk', 'i18n', 'version', 'experimental'];
+    const skips = ['style', 'core', 'locale', 'cdk', 'i18n', 'version', 'experimental', 'node_modules'];
```
Comment (hsuanxyz) [inline scripts/site/generate-site.js:44]:
```suggestion
    const skips = ['style', 'core', 'locale', 'cdk', 'i18n', 'version', 'experimental'];
```

**Review by hsuanxyz — CHANGES_REQUESTED** [review]

File: components/core/overlay/nz-connected-overlay.ts (hunk 2 above)
Comment (hsuanxyz) [reply components/core/overlay/nz-connected-overlay.ts:39]: https://github.com/angular/components/blob/59002e1649/src/cdk/overlay/overlay-directives.ts#L397
https://github.com/angular/components/blob/59002e1649123922df3532f4be78c485a73c5bc1/src/cdk/overlay/position/flexible-connected-position-strategy.ts#L1138

**Review by wendellhu95 — APPROVED** [review]
LGTM

**Review by hsuanxyz — APPROVED** [review]
LGTM

**Comment by RoboZoom** [comment]
It's fantastic that this is merged, but the latest release on the home page still is version 12.1.1 and references Angular v12.  Are you planning on updating the release version number for this?

Thanks for the work!

**Comment by FOWind** [comment]
When it will release after merged?

**Comment by hsuanxyz** [comment]
@FOWind https://github.com/NG-ZORRO/ng-zorro-antd/pull/7158

**Pull request merged**